app/javascript/components/UserProfileMenu/index.jsx

```diff
@@ -0,0 +1,144 @@
+      console.log(changes.selectedItem)
+      setIsOpen(
+        changes.selectedItem === 'default-office' ||
+        changes.selectedItem === 'language-settings' ||
+        changes.selectedItem === 'general-settings' ||
+        changes.isOpen
+      )
+    }
+
+    if (R.hasPath(['selectedItem'])(changes)) {
+      const itemSelected = R.prop('selectedItem')(changes)
+      if (itemSelected === 'general-settings') {
+        switch (tempSelectedItem) {
+          case 'default-office':
+            stateAndHelpers.setHighlightedIndex(0)
+            break
+          case 'language-settings':
+            stateAndHelpers.setHighlightedIndex(1)
+        }
+      }
+      setTempSelectedItem(itemSelected)
+    }
+  }
+
+  const handleSelect = selectedItem => {
+    if (tempSelectedItem === 'default-office') {
+      if (R.hasPath(['office'])(selectedItem)) {
+        handleOfficeSelect(selectedItem.office)
+        setSelectedItem(selectedItem)
+      }
+    } else if (tempSelectedItem === 'language-settings') {
+      if (R.hasPath(['language'])(selectedItem)) {
+        i18n.changeLanguage(selectedItem.language.value, () => {
+          // TODO: Handle callback (error/success)
+        })
+        setSelectedItem(selectedItem)
+      }
+    } else if (selectedItem === 'admin') {
+      router.push('/portal/admin')
+    } else if (selectedItem === 'home') {
+      router.push('/portal')
+    } else if (selectedItem === 'sign-out') {
+      router.push('/portal/sign_out')
+    }
+  }
+
+  const renderItems = () => {
+    if (tempSelectedItem === 'default-office') {
+      return <DefaultOfficeMenu offices={offices} previousMenuValue="general-settings" selectedItem={selectedItem} t={t} />
+    } else if (tempSelectedItem === 'language-settings') {
+      return <LanguageMenu languages={languages} previousMenuValue="general-settings" selectedItem={selectedItem} t={t} />
+    } else {
+      return <GeneralSettingsMenu menuValues={{ defaultOffice: 'default-office', languageSettings: 'language-settings'}} isAdmin={currentUser.isAdmin} pathname={location.pathname} t={t} />
+    }
+  }
+
+  return (
+    R.isNil(currentUser) || R.isEmpty(currentUser) ? null :
+    <Dropdown
+      selectedItem={selectedItem}
+      // This is used to detect what items are selected with a check mark.
+      downshiftProps={{
+        itemToString: item => (item.office && item.office.id) + (item.language && item.language.value),
+      }}
+      isOpen={isOpen}
+      onStateChange={handleStateChange}
+      onSelect={handleSelect}
+    >
+      <Trigger>
+        <UserProfileContainer>
+          <Avatar>
+            <img src={currentUser.photo} alt="User Avatar" />
+          </Avatar>
+          <UserDetails>
+            <UserName>{currentUser.name}</UserName>
+            <MD>{currentUser.office && currentUser.office.name}</MD>
+          </UserDetails>
+        </UserProfileContainer>
+      </Trigger>
+      <GardenMenu hasArrow>{renderItems()}</GardenMenu>
```

Comment:
you don't need to change it, but you could use ramda here:
```
<GardenMenu hasArrow>{
        R.cond([
          [R.equals('default-office'). R.always(<DefaultOfficeMenu offices={offices} previousMenuValue="general-settings" selectedItem={selectedItem} t={t} />)],
          [R.equals('language-settings'), R.always(<LanguageMenu languages={languages} previousMenuValue="general-settings" selectedItem={selectedItem} t={t} />)],
          [R.T, R.always(<GeneralSettingsMenu menuValues={{ defaultOffice: 'default-office', languageSettings: 'language-settings'}} isAdmin={currentUser.isAdmin} pathname={location.pathname} t={t} />)]
        ])(tempSelectedItem)
      }
</GardenMenu>
```

Reply:
might be great to use this if we needed to compose the boolean condition with something more complex 👍